src/nix.rs

```diff
@@ -6,6 +6,25 @@ pub struct PrefetchInfo {
     hash: String,
 }
 
+pub async fn hash_to_sri(hash_type: &str, hash: String) -> Result<String> {
+    let output = tokio::process::Command::new("nix")
```

Comment:
Will this require a specific minimum nix version? e.g. 2.4 and newer?

Reply:
Only if we shell out, if we use Tvix, not really.

Reply:
If tvix manages to publish these operations in some useable format. I've some random bits and pieces from other nix-related projects that do this. Since I also want to remove reliance on `nix-prefetch-url` entirely and add that to npins.